volume/volume.go

```diff
@@ -324,6 +328,10 @@ type BlockDriver interface {
 	// Detach device from the host.
 	// Errors ErrEnoEnt, ErrVolDetached may be returned.
 	Detach(ctx context.Context, volumeID string, options map[string]string) error
+	// ControllerPublish csi extension
```

Comment:
I am not in favor of adding ControllerPublish and Unpublish functions to the BlockDriver interface, since they are just a combination of Attach and Detach.

Reply:
These changes are needed for compilation support. A driver needs to support all interfaces.

Reply:
If you invoke driver's Attach/Detach in the Controller Publish/Unpublish implementation then you don't need this.

Reply:
will revisit this with @pnookala-px